---
title: "1-Preprocessing_workflow"
author: "Joan Abinet"
date: "`r format (Sys.time(), format='%Y-%m-%d %H:%M:%S %z')`"
output:
  pdf_document:
    toc: true
    toc_depth: 2
  header-includes:
    -|
    ```{=latex}
    \usepackage{fvextra}
    \DefineVerbatimEnvironment{Highlighting}{Verbatim}{
      breaksymbolleft={}, 
      showspaces = false,
      showtabs = false,
      breaklines,
      commandchars=\\\{\}
    }
    ```
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(tidy.opts = list(width.cutoff = 65), tidy = TRUE)
```

\newpage

# Introduction

Lung endothelial cells (CD45-, EpCAM-, CD31+) were FACS-sorted as living singlet from lung single-cell suspensions pooled from 4 mice per condition: control, d3 a-Ly6G, d3 a-CXCR2, d14 a-Ly6G, d14 a-CXCR2. For each sample, an aliquot of Trypan blue-treated cells was examined under the microscope for counting, viability and aggregate assessment following FACS sorting. Viability was above 90% for all samples and no aggregates were observed. Cell preparations were centrifuged and pellets were resuspended in calcium- and magnesium-free PBS containing 0.4 mg ml-1 UltraPure BSA (Thermo Fisher Scientific)

Single-cell RNA sequencing was performed using the 10x Genomics Chromium Next GEM Single Cell 3’ v3.1 kit. A total of 13,000 cells were loaded per sample, with eight samples processed on a single chip using the Chromium Controller instrument. The manufacturer’s protocol was strictly followed throughout the procedure. cDNA quality and fragment size distribution were assessed using the Agilent High Sensitivity DNA Kit on a Fragment Analyzer. Final libraries were quality-checked on a QIAxcel system (Qiagen), quantified by qPCR (Roche), and pooled in equimolar amounts.

Sequencing was performed on an Illumina NovaSeq 6000 system using
an S4 flow cell with paired-end 2×150 bp reads (Read1: 150 cy, read2: 150 cy, index1:10cy, index2: 10cy), targeting a depth of 30,000 reads per cell for each sample. Raw sequencing data were demultiplexed using sample-specific indexes, quality-filtered, and converted into FASTQ files using bcl2fastq. Sequencing data quality was evaluated per sample using FastQC, and summarized across all samples using MultiQC.

The Cell Ranger (v6.1.2) application (10x Genomics) was then used to demultiplex the BCL files into FASTQ files (cellranger mkfastq), to perform alignment (using Cell Ranger human genome references  6.1.2 GRCm38/release 102), filtering and unique molecular identifier counting and to produce gene-barcode matrices (cellranger count).

# Loading packages
```{r}
suppressMessages({
  library(dplyr)
  library(Seurat)
  library(patchwork)
  library(ggplot2)
  library(formatR)
  library(SingleR)
})
```

# Loading data
The count files can be download from GEO
```{r, message=FALSE, warning=FALSE, results= "hide" }
# Specify your 10x directory
all_dirs <- list.dirs(path = "Count", full.names = TRUE, recursive = F)

list_sample_name <- c("aCXCR2-D14", "aCXCR2-D3", "aLy6G-D14", "aLy6G-D3", "PBS-PBS")

list_sample <- list()
for (i in 1:length(all_dirs)) {
  
  Seq_raw_file <- Read10X(data.dir = all_dirs[i])
  Seurat_file <- CreateSeuratObject(counts = Seq_raw_file, project = list_sample_name[i], min.cells = 3, min.features =   200)
  list_sample <- append(list_sample, Seurat_file)
}
list_sample

endothelial_cells <- merge(list_sample[[1]], y = list_sample[-1], add.cell.ids = c("1","2", "3", "4", "5"), project = "Endothelial_Cells")
```

# Quality Control
```{r, fig.width=6, fig.height=4}
endothelial_cells[["percent.mt"]] <- PercentageFeatureSet(endothelial_cells, pattern = "^mt-")
VlnPlot(endothelial_cells, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
```

# Filtering
```{r, fig.width=6, fig.height=4}
endothelial_cells <- subset(endothelial_cells, subset = nFeature_RNA > 200 & nFeature_RNA < 4000  & percent.mt < 15) 
p1 <- VlnPlot(endothelial_cells, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, pt.size = 0)
p1
```

# Pre-processing workflow
```{r, message=FALSE, warning=FALSE, results= "hide" }
# Data Normalization
endothelial_cells <- NormalizeData(endothelial_cells, normalization.method = "LogNormalize", scale.factor = 10000)

# Feature selection
endothelial_cells <- FindVariableFeatures(endothelial_cells, selection.method = "vst", nfeatures = 2000)
top10 <- head(VariableFeatures(endothelial_cells), 10)
p2 <- VariableFeaturePlot(endothelial_cells)
p2 <- LabelPoints(plot = p2, points = top10, repel = TRUE)

# Scaling the data
all.genes <- rownames(endothelial_cells)
endothelial_cells <- ScaleData(endothelial_cells, features = all.genes)

# Linear dimensional reduction
endothelial_cells <- RunPCA(endothelial_cells, features = VariableFeatures(object = endothelial_cells))
p3 <- DimPlot(endothelial_cells, reduction = "pca")

# Determining the ‘dimensionality’ of the dataset
p4 <- ElbowPlot(endothelial_cells)
```

```{r, fig.width=15, fig.height=6 }
p2 + p3 + p4
```


# Clustering
```{r, message=FALSE, warning=FALSE, results= "hide" }
endothelial_cells <- FindNeighbors(endothelial_cells, dims = 1:15)
endothelial_cells <- FindClusters(endothelial_cells, resolution = 0.25)

endothelial_cells <- RunUMAP(endothelial_cells, dims = 1:15)
```

```{r}
DimPlot(endothelial_cells, reduction = "umap", label = T)
#ggsave("Umap_1.png", height = 8, width=12)
```

# Endothelial genes
```{r, fig.width=10, fig.height=10}
FeaturePlot(endothelial_cells, features = c("Vwf","Mcam","Rgs5", "Pecam1"))
```


```{r}
DimPlot(endothelial_cells,group.by = "orig.ident", reduction = "umap")
```

# Automatic annotation

## Annotation with TabulaMurisData
```{r, message=FALSE, warning=FALSE, results= "hide"}
library(ExperimentHub)
require(scuttle)

eh <- ExperimentHub()
query(eh, "TabulaMurisData")
ref <- eh[['EH1617']]

lung_ref <-ref[,!is.na(ref$cell_ontology_class)]
lung_ref <- lung_ref[,lung_ref$tissue == "Lung"]

lung_ref <-logNormCounts(lung_ref)

tested_data <- as.SingleCellExperiment(endothelial_cells)
tested_data <- logNormCounts(tested_data)

results <- SingleR(test =tested_data  , ref = lung_ref, labels = lung_ref$cell_ontology_class)
cell_annotations <- results
table(cell_annotations$labels)

rownames(endothelial_cells@meta.data)
rownames(cell_annotations)
setequal(rownames(endothelial_cells@meta.data), rownames(cell_annotations))

endothelial_cells[["singleR_annotation"]] <- cell_annotations[,c(4)]
```

```{r}
DimPlot(endothelial_cells, reduction = "umap", group.by = "singleR_annotation", cols = c())
#ggsave("Umap_singleR1.png", height = 8, width=12)
```

## Annotation with MouseRNAseqData
```{r, message=FALSE, warning=FALSE, results= "hide"}
library(celldex)

ref.bulk <- MouseRNAseqData()

results_celldex <- SingleR(test = tested_data, ref = ref.bulk, labels = ref.bulk$label.main)
endothelial_cells[["singleR_annotation2"]] <- results_celldex[,c(4)]
```

```{r}
DimPlot(endothelial_cells, reduction = "umap", group.by = "singleR_annotation2")
#ggsave("Umap_singleR2.png", height = 8, width=12)
```

Clusters 7, 11 and 13 were identified as contamination were removed.
```{r, eval=FALSE}
endothelial_cells <- subset(endothelial_cells, seurat_clusters %in% c(7,11,13), invert=T)
saveRDS(endothelial_cells, "endothelial_cells.rds")
```


```{r}
sessionInfo()
```



